segmentstore/server/src/test/java/io/pravega/segmentstore/server/attributes/AttributeIndexTests.java

```diff
@@ -627,6 +631,68 @@ public void testRecoveryAfterIncompleteUpdate() {
         checkIndex(idx2, expectedValues);
     }
 
+    /**
+     * Tests the ability of the Attribute Index to recover correctly after an update has been successfully written to Storage,
+     * but the previous value of the {@link Attributes#ATTRIBUTE_SEGMENT_ROOT_POINTER} has been truncated out without
+     * having the new value persisted (most likely due to a system crash).
+     * In this case, the {@link Attributes#ATTRIBUTE_SEGMENT_ROOT_POINTER} value should be ignored and the index should
+     * be attempted to be read from Storage without providing hints to where to start reading from.
+     */
+    @Test
+    public void testTruncatedRootPointer() {
+        val attributeSegmentName = NameUtils.getAttributeSegmentName(SEGMENT_NAME);
+        val config = AttributeIndexConfig
+                .builder()
+                .with(AttributeIndexConfig.MAX_INDEX_PAGE_SIZE, 1024)
+                .with(AttributeIndexConfig.ATTRIBUTE_SEGMENT_ROLLING_SIZE, 8)
+                .build();
+        final int attributeCount = 20;
+        val attributes = IntStream.range(0, attributeCount).mapToObj(i -> new UUID(i, i)).collect(Collectors.toList());
+        @Cleanup
+        val context = new TestContext(config);
+        populateSegments(context);
+
+        // 1. Populate and verify first index.
+        val expectedValues = new HashMap<UUID, Long>();
+        long nextValue = 0;
+        long previousRootPointer = -1;
+        boolean invalidRootPointer = false;
+        for (UUID attributeId : attributes) {
+            val idx = context.index.forSegment(SEGMENT_ID, TIMEOUT).join();
+
+            val value = nextValue++;
+            expectedValues.put(attributeId, value);
+            val updateBatch = Collections.singletonMap(attributeId, value);
+
+            val rootPointer = idx.update(updateBatch, TIMEOUT).join();
+            val startOffset = context.storage.getStreamSegmentInfo(attributeSegmentName, TIMEOUT).join().getStartOffset();
+            if (previousRootPointer >= 0) {
+                context.containerMetadata.getStreamSegmentMetadata(SEGMENT_ID)
+                        .updateAttributes(Collections.singletonMap(Attributes.ATTRIBUTE_SEGMENT_ROOT_POINTER, previousRootPointer));
+                invalidRootPointer |= previousRootPointer < startOffset;
```

Comment:
A comment explaining how root pointer is always set to a previous value to simulate scenario and where truncation happens would be helpful.

Reply:
Thanks for adding comment.